### PR TITLE
BrickColor support

### PR DESCRIPTION
**Pull request by Kampfkarren**
BrickColors are serialized as normal numbers, so this is the best way to do it I feel.

Makes use of TryFrom<T>, stabilized in Rust 1.34.0.

### CODE REVIEW

File: rbx_dom_weak/src/brick_color.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			fn try_from(value: u8) -> Result<Self, Self::Error> {
+				match value {
+					$(
+						$value => Ok(BrickColor::$enum),
+					)+
+
+					_ => Err(()),
+				}
+			}
+		}
```
Comment (LPGhatguy) [inline rbx_dom_weak/src/brick_color.rs]: I'm not super comfortable bumping the minimum Rust version for `TryFrom`, can we just use an inherent impl with a descriptive name?

Comment (Kampfkarren) [reply rbx_dom_weak/src/brick_color.rs]: Done.

File: rbx_dom_weak/src/brick_color.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+macro_rules! make_brick_color {
+	({$([$enum:ident, $name:tt, $value:tt],)+}) => {
+		#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+		pub enum BrickColor {
+			$($enum = $value,)+
+		}
+
+		impl BrickColor {
+			pub fn from_palette(value: u8) -> Option<BrickColor> {
+				match value.into() {
```
Comment (LPGhatguy) [inline rbx_dom_weak/src/brick_color.rs]:
```suggestion
				match value {
```

**Comment by LPGhatguy** [comment]
Thanks a bunch!

**Pull request merged**